core/server/fluxruntime.go

```diff
@@ -247,7 +247,7 @@ func (cs *coreServer) GetReconciledObjects(ctx context.Context, msg *pb.GetRecon
 }
 
 func (cs *coreServer) GetChildObjects(ctx context.Context, msg *pb.GetChildObjectsRequest) (*pb.GetChildObjectsResponse, error) {
-	clustersClient, err := cs.clustersManager.GetImpersonatedClientForCluster(ctx, auth.Principal(ctx), msg.ClusterName)
+	clustersClient, err := cs.clustersManager.GetImpersonatedClient(ctx, auth.Principal(ctx))
```

Comment:
This seems like a logic change that I wouldn't expect. We don't need to get a client scoped for a cluster?

Reply:
After reading more, I think I understand. Should we remove the `GetImpersonatedClientForCluster` method? Is it used anywhere else?

Reply:
yep, I'll remove it.

Reply:
`GetImpersonatedClientForCluster` is used a little bit in EE for querying details pages.

It will be slightly slower to build up the client set for these cases, mostly if another cluster is unreachable in a funny way (timeout's ~8s).

Reply:
@foot Any idea if we can control that timeout?

Reply:
The goal of the PR is not to have to build clients every time, I wonder what happens if we keep the clients around for longer than the 5 min we have now, and as Jordan suggested, decrease to a more sensible number.

Reply:
Why do we want to remove `GetImpersonatedClientForCluster` at all?

I would not use a ttl cache at all, because the resource requirements aren't possible to control - unless there's issues with connections breaking at a certain time, I'd use an LRU.

Reply:
> @foot Any idea if we can control that timeout?

- We first try to connect to a leaf in `ClientConfigWithUser`, it will apply the `DefaultKubeConfigOptions` one of which is `WithFlowControl`. This does a `HEAD` request to the leaf which can take 5s to hit `kubeClientDialTimeout`. So most unreachable leafs will stop here.
- Then when we `client.New` the client will start up and fill in the restMapper by querying CRDs, this has has 8s to hit the `kubeClientTimeout` I think...

So... yeah, if everything goes terribly it can at least take 4.9s + 7.9s. Both those timeout values are in factory.go to tweak.

Reply:
> I wonder what happens if we keep the clients around for longer than the 5 min we have now

- We'd need to wipe out the cache if clusters change. There is a `chan` we can subscribe to now though.
- If 1 cluster is briefly unreachable then it wouldn't be in the client set until we try again in 5mins.

Reply:
> So... yeah, if everything goes terribly it can at least take 4.9s + 7.9s. Both those timeout values are in factory.go to tweak.

Because it's better to have a single cached client for each cluster, and `GetImpersonatedClientForCluster` was kind of a hack to overcome the slow client creation problem. If we can fix that we should remove it. And when querying a specific cluster we should use the `ClusterName` param on the calls.

Reply:
> Because it's better to have a single cached client for each cluster

That method should grab the cached client, yeah.

> If we can fix that we should remove it.

But, as @foot explained, clusters come and go, so this change won't affect any of the problems this method works around.

(Which in my opinion suggests `GetImpersonatedClient` doesn't work and `GetImpersonatedClientForCluster` is the one we should keep, and the decision which cluster to query/poll should live together with the decision of when to poll. That would let us easily exclude broken clusters, back off slow clusters, render partial results as they come in until we have responses from all clusters, etc. But I digress)